packages/std/src/math/signed_decimal_256.rs

```diff
@@ -0,0 +1,3226 @@
+        assert_eq!(
+            SignedDecimal256::from_str("0.123").unwrap(),
+            SignedDecimal256::permille(123)
+        );
+
+        assert_eq!(
+            SignedDecimal256::from_str("40.00").unwrap(),
+            SignedDecimal256::percent(4000)
+        );
+        assert_eq!(
+            SignedDecimal256::from_str("04.00").unwrap(),
+            SignedDecimal256::percent(400)
+        );
+        assert_eq!(
+            SignedDecimal256::from_str("00.40").unwrap(),
+            SignedDecimal256::percent(40)
+        );
+        assert_eq!(
+            SignedDecimal256::from_str("00.04").unwrap(),
+            SignedDecimal256::percent(4)
+        );
+        // Negative decimals
+        assert_eq!(
+            SignedDecimal256::from_str("-00.04").unwrap(),
+            SignedDecimal256::percent(-4)
+        );
+        assert_eq!(
+            SignedDecimal256::from_str("-00.40").unwrap(),
+            SignedDecimal256::percent(-40)
+        );
+        assert_eq!(
+            SignedDecimal256::from_str("-04.00").unwrap(),
+            SignedDecimal256::percent(-400)
+        );
+
+        // Can handle DECIMAL_PLACES fractional digits
+        assert_eq!(
+            SignedDecimal256::from_str("7.123456789012345678").unwrap(),
+            SignedDecimal256(Int256::from(7123456789012345678i128))
+        );
+        assert_eq!(
+            SignedDecimal256::from_str("7.999999999999999999").unwrap(),
+            SignedDecimal256(Int256::from(7999999999999999999i128))
+        );
+
+        // Works for documented max value
+        assert_eq!(
+            SignedDecimal256::from_str(
+                "57896044618658097711785492504343953926634992332820282019728.792003956564819967"
+            )
+            .unwrap(),
+            SignedDecimal256::MAX
+        );
+        // Works for documented min value
+        assert_eq!(
+            SignedDecimal256::from_str(
+                "-57896044618658097711785492504343953926634992332820282019728.792003956564819968"
+            )
+            .unwrap(),
+            SignedDecimal256::MIN
+        );
+        assert_eq!(
+            SignedDecimal256::from_str("-1").unwrap(),
+            SignedDecimal256::negative_one()
+        );
+    }
+
+    #[test]
+    fn signed_decimal_256_from_str_errors_for_broken_whole_part() {
+        match SignedDecimal256::from_str("").unwrap_err() {
+            StdError::GenericErr { msg, .. } => assert_eq!(msg, "Error parsing whole"),
+            e => panic!("Unexpected error: {e:?}"),
+        }
+
+        match SignedDecimal256::from_str(" ").unwrap_err() {
+            StdError::GenericErr { msg, .. } => assert_eq!(msg, "Error parsing whole"),
+            e => panic!("Unexpected error: {e:?}"),
+        }
+
+        match SignedDecimal256::from_str("-").unwrap_err() {
+            StdError::GenericErr { msg, .. } => assert_eq!(msg, "Error parsing whole"),
+            e => panic!("Unexpected error: {e:?}"),
+        }
+    }
```

Comment:
I would change it to:
```
    #[test]
    fn signed_decimal_256_from_str_errors_for_broken_whole_part() {
        let expected_err = StdError::generic_err("Error parsing whole");
        assert_eq!(expected_err, SignedDecimal256::from_str("").unwrap_err());
        assert_eq!(expected_err, SignedDecimal256::from_str(" ").unwrap_err());
        assert_eq!(expected_err, SignedDecimal256::from_str("-").unwrap_err());
    }
```
because the `e => panic![...]` arms have no chance to be executed in tests.
If someone accidentally changes the error type, the proposed version will also catch it.

All similar suggested changes are placed also in branch https://github.com/CosmWasm/cosmwasm/tree/1711-signed-decimal256-coverage for convinience.

Final code-coverage measured with grcov is 100%. Tarpaulin is also 100% for this file.